src/store/staking/stakingSlice.ts

```diff
@@ -60,8 +89,19 @@ export const stakingSlice = createSlice({
       newStake.tStake = stakeType === StakeType.T ? _amount : "0"
       newStake.totalInTStake = _amount
 
+      const _isBeforeBonusDeadline = isBeforeBonusDeadline()
+      newStake.bonusEligibility = {
+        eligibleStakeAmount: _isBeforeBonusDeadline ? _amount : "0",
+        hasPREConfigured: false,
+        hasActiveStake: _isBeforeBonusDeadline,
+        hasUnstakeAfterBonusDeadline: false,
+        reward: calculateStakingBonusReward(_amount),
+      }
+
       state.stakes = [newStake, ...state.stakes]
```

Comment:
Wouldn't it be better to add this `newStake` at the end? I think the stakes are show in chronilogical order when fetched from the chain.

```suggestion
      state.stakes = [...state.stakes, newStake]
```

Reply:
Tbh from my point of view is better to see the added record at the top of the list. So we should probably reverse the staked events array here https://github.com/threshold-network/token-dashboard/blob/main/src/hooks/useFetchOwnerStakes.ts#L28.

Reply:
6c1ec13